src/main/java/com/mate/lib/EmailValidator.java

```diff
@@ -0,0 +1,18 @@
+package com.mate.lib;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
+
+    public static final String EMAIL_VALIDATION_REGEX = "^(.+)@(.+)$";
+
+    @Override
+    public void initialize(ValidEmail constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(String field, ConstraintValidatorContext context) {
+        return field.matches(EMAIL_VALIDATION_REGEX);
```

Comment:
Remember, that null.equals(smth) == NPE. HINT: be careful with implementation of the isValid() method.